src/test/java/loanbook/model/ModelManagerTest.java

```diff
@@ -63,6 +65,23 @@ public void hasLoan_loanInLoanBook_returnsTrue() {
         assertTrue(modelManager.hasLoan(ALICE));
     }
 
+    @Test
+    public void getLoanById_loanExists_success() {
+        modelManager.addLoan(ALICE);
+        Optional<Loan> retrievedLoan = modelManager.getLoanById(ALICE.getLoanId());
+
+        assertTrue(retrievedLoan.isPresent());
+        assertEquals(retrievedLoan.get(), ALICE);
```

Comment:
Will not throw an error as the above line `assertTrue(retrievedLoan.isPresent());` will short-circuit the test if retrievedLoan is not present.

Reply:
That's indeed correct.

And if `retrievedLoan.get()` threw an `AssertionError`, that means the test is failing and something needs to be fixed.